### PR TITLE
Add an option to disable search for homeservers which may not be interested in it

### PR DESCRIPTION
**Pull request by turt2live**
This is useful for homeservers not intended for users, such as bot-only homeservers or ones that only process IoT data.

This was done with a community hat on:
Signed-off-by: Travis Ralston <travis@t2bot.io>

### CODE REVIEW

File: synapse/storage/search.py [hunk 1]
```diff
@@ -45,27 +45,29 @@ class SearchStore(BackgroundUpdateStore):
...
 
-        # we used to have a background update to turn the GIN index into a
-        # GIST one; we no longer do that (obviously) because we actually want
-        # a GIN index. However, it's possible that some people might still have
-        # the background update queued, so we register a handler to clear the
-        # background update.
-        self.register_noop_background_update(
-            self.EVENT_SEARCH_USE_GIST_POSTGRES_NAME,
-        )
+        if hs.config.enable_search:
```
Comment (hawkowl) [inline synapse/storage/search.py]: You could change this into a `if not hs.config.enable_search: return` to avoid a big diff here.

Comment (turt2live) [reply synapse/storage/search.py]: yes, although I didn't really want to risk people adding things to the end and wondering why they aren't be called in the future. Happy to switch it around though.

Comment (hawkowl) [reply synapse/storage/search.py]: We'll be able to catch cases like that with tests, I hope :)

Comment (turt2live) [reply synapse/storage/search.py]: done

**Pull request merged**